src/storage/src/source/mysql.rs

```diff
@@ -223,6 +216,15 @@ impl SourceRender for MySqlSourceConnection {
     }
 }
 
+#[derive(Clone, Debug)]
+struct SubsourceInfo {
+    output_index: usize,
+    name: MySqlTableName,
+    desc: MySqlTableDesc,
+    initial_gtid_set: String,
+    resume_upper: Antichain<GtidPartition>,
+}
+
```

Comment:
I introduced this helper struct to combine the `subsource_resume_uppers` and `table_info` parameters that were previously being passed to both the replication & snapshot `render` functions.